### PR TITLE
Fix pagebreak to check

### LINKED ISSUE #2473: Pagebreak to does not behave correctly (again)
### Description

This example code:
```typ
#set page(paper: "a8", numbering: "1")

= Part 1
#lorem(45)

#pagebreak(to: "odd", weak: true)
= Part 2
#lorem(20)
```
In 0.8.0 renders:
image

But after #2182:
image

It seems like solving one thing broke another 😭 

### Reproduction URL

_No response_

### Operating system

Windows

### Typst version

- [X] I am using the latest version of Typst

### PR DESCRIPTION
**Pull request by Zheoni**
After #2182 the parity check was inverted because the empty frame was added to the end of the old styles (I think). This introduced the problem where the page counter was being checked with the current number of pages and not taking into account the pages that were about to be appended. This problem was not noticeable when between pagebreaks the content didn't introduce more than one page. This PR fixes the parity check and adds a new entry to the existing test. Also, fixes #2473.

This is my first time fiddling with typst source code and although it's a small PR, please be careful when reviewing it. Sorry in advance if I broke anything 😓.

### CODE REVIEW

**Comment by Zheoni** [comment]
There's still (at least) one problem this does not fix: a `pagebreak(to: "even", weak: true)` at the beginning of the document, when the first page is empty, will not work and won't add an empty page:

```typ
#set page(paper: "a8", numbering: "1")

#pagebreak(to: "even", weak: true)

Second
```

Renders (also since #2182):
image

And it should render (as it does in 0.8.0):
image

I have been trying to understand the layout/resolution code and I think it's because weak pagebreaks are removed before the empty pages are added to match parity. That removes the page with the `clear_to` style and therefore it's never added? Maybe my reasoning is wrong, but the issue is there.

However, this PR still solves the original issue. I now don't know if the correct path is to continue working in solving this new thing in this PR or open a new issue.

File: tests/typ/layout/pagebreak-parity.typ [hunk 1]
```diff
@@ -21,3 +21,15 @@ Seventh
...
+#set page(height: 30pt, width: 80pt)
+
+// Test when content extends to more than one page
+First
+
+Second
+
+#pagebreak(to: "odd")
+
+Third
```
Comment (PgBiel) [inline tests/typ/layout/pagebreak-parity.typ:35]: nit: Newline here please :)

**Review by PgBiel — COMMENTED** [review]
I think it's fair to merge this PR first and then make a separate issue/PR pair for the other problem.

**Comment by PgBiel** [comment]
In principle (from a brief analysis including the surrounding code), LGTM.

cc @laurmaedje for further opinions

**Comment by laurmaedje** [comment]
I will take a closer look later, but this does make sense.

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**